common.go

```diff
@@ -300,33 +263,20 @@ func SingleVersionValidator(desiredVersion Version) VersionValidator {
 	}
 }
 
-// checkSignatureState sanity-checks some signature parameters. When
-// called by the signer, a non-nil error should cause a panic, but
-// when called by the verifier, it should be treated as a regular
-// error.
-func checkSignatureState(version Version, chunk []byte, isFinal bool) error {
-	makeErr := func() error {
-		return fmt.Errorf("invalid signature state: version=%s, len(chunk)=%d, isFinal=%t", version, len(chunk), isFinal)
-	}
-
+func checkChunkState(version Version, chunkLen int, blockIndex uint64, isFinal bool) error {
 	switch version.Major {
 	case 1:
-		if (len(chunk) == 0) != isFinal {
-			return makeErr()
+		// For V1, we derive isFinal from the chunk length, so
+		// if there's a mismatch, that's a bug and not a
+		// stream error.
+		if (chunkLen == 0) != isFinal {
+			panic(fmt.Sprintf("chunkLen=%d and isFinal=%t", chunkLen, isFinal))
 		}
 
 	case 2:
-		// With V2, it's valid to have a final packet with
-		// non-empty chunk, so the below is the only remaining
-		// invalid state.
-		//
-		// TODO: Ideally, we'd disallow empty packets even
-		// with isFinal set, but we still want to allow
-		// signing an empty message. Plumb through an isFirst
-		// flag and change "!isFinal" to "!isFirst ||
-		// !isFinal".
-		if (len(chunk) == 0) && !isFinal {
-			return makeErr()
+		// TODO: Ideally, we'd have tests exercising this case.
```

Comment:
I can perhaps see how to do this with encrypt and sign, i.e. use the tweakable encoders to emit an empty packet, but I'm not sure how to do it for signcryption. Ideas?